pallets/passkey/src/lib.rs

```diff
@@ -131,28 +145,42 @@ pub mod module {
 	#[pallet::validate_unsigned]
 	impl<T: Config> ValidateUnsigned for Pallet<T>
 	where
-		<T as frame_system::Config>::RuntimeCall: Dispatchable<Info = DispatchInfo>,
+		<T as frame_system::Config>::RuntimeCall:
+			From<Call<T>> + Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
 	{
 		type Call = Call<T>;
 		fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
 			let payload = Self::filter_valid_calls(&call)?;
 			Self::validate_signatures(&payload)?;
-
+			let tx_charge = ChargeTransactionPayment::<T>(
+				payload.passkey_call.account_id.clone(),
+				call.clone(),
+			);
+			tx_charge.validate()?;
 			let nonce_check = PasskeyNonce::new(payload.passkey_call.clone());
 			nonce_check.validate()
 		}
 
 		fn pre_dispatch(call: &Self::Call) -> Result<(), TransactionValidityError> {
+			let payload = Self::filter_valid_calls(&call)?;
+			let tx_charge = ChargeTransactionPayment::<T>(
+				payload.passkey_call.account_id.clone(),
```

Comment:
I suggest swapping the order of these two dispatch calls:

1.	Filter valid calls
2.	Validate signatures
3.	Increase nonce
4.	Payment

It seems that we are calling validate_unsigned inside pre_dispatch, which might be redundant. The order reflects the order of our [SignedExtra](https://github.com/frequency-chain/frequency/blob/main/runtime/frequency/src/lib.rs#L290).

Reply:
Fixed the order for both `pre_dispatch` and `validate_unsigned`